### PR TITLE
replace api version check with specific feature checks

### PR DESCRIPTION
**Pull request by acjohnson**
* ensure compat with hide_version Grafana setting

### CODE REVIEW

**Comment by acjohnson** [comment]
Fixes #94

**Comment by acjohnson** [comment]
This also fixes issues with missing `uid` dashboard and folder keys in Grafana version < 5.0.0

**Comment by acjohnson** [comment]
@ysde let me know if you have any questions

**Comment by acjohnson** [comment]
@Mikey1993 any comments on this as it relates to https://github.com/ysde/grafana-backup-tool/issues/94#issuecomment-751502021

**Review by Mikey1993 — CHANGES_REQUESTED** [review]
@acjohnson ,
I suggest changing the `paging_feature_check` to be something like this:

```
def paging_feature_check(grafana_url, http_get_headers, verify_ssl, client_cert, debug):
    def get_first_dashboard_by_page(page):
        (status, content) = search_dashboard(page, 1, grafana_url, http_get_headers, verify_ssl, client_cert, debug)
        if status == 200 and len(content):
            dashboard_values = sorted(content[0].items(), key=lambda kv: str(kv[1]))
            return True, dashboard_values
        else:
            if len(content):
                return "get dashboards failed, status: {0}, msg: {1}".format(status, content)
            else:
                # No dashboards exist, disable paging feature
                return False

    # Get first dashboard on first page
    (status, content) = get_first_dashboard_by_page(1)
    if status is False:
        return False  # Paging feature not supported
    elif status is True:
        dashboard_one_values = content
    else:
        return content  # Fail Message

    # Get second dashboard on second page
    (status, content) = get_first_dashboard_by_page(2)
    if status is False:
        return False  # Paging feature not supported
    elif status is True:
        dashboard_two_values = content
    else:
        return content  # Fail Message

    # Compare both pages
    return dashboard_one_values != dashboard_two_values
```

**Comment by acjohnson** [comment]
@Mikey1993 I tested and corrected your stylistic refactor of `paging_feature_check` and added it to this PR

**Comment by acjohnson** [comment]
@ysde any heartburn with merging this one?

File: grafana_backup/save_folders.py [hunk 1]
```diff
@@ -46,17 +47,22 @@ def save_folder_setting(folder_name, file_name, folder_settings, folder_path, pr
...
-def get_individual_folder_setting_and_save(folders, folder_path, log_file, grafana_url, http_get_headers, verify_ssl, client_cert, debug, pretty_print):
+def get_individual_folder_setting_and_save(folders, folder_path, log_file, grafana_url, http_get_headers, verify_ssl, client_cert, debug, pretty_print, uid_support):
     file_path = folder_path + '/' + log_file
     with open(u"{0}".format(file_path), 'w+') as f:
         for folder in folders:
-            (status, content) = get_folder(folder['uid'], grafana_url, http_get_headers, verify_ssl, client_cert, debug)
+            if uid_support:
+                folder_uri = "uid/{0}".format(folder['uid'])
+            else:
+                folder_uri = folder['uri']
```
[the remark this answers is not in the document]
Comment (acjohnson) [reply grafana_backup/save_folders.py:57]: @ysde fair question but no this does not affect the backup or restore function. Backups created using previous versions of `grafana-backup-tool` will still be valid.

Also note that `uid_support` is available on all Grafana versions 5.0.0 and newer

**Pull request merged**